From 5379ea0d6503c5aac14ae8c174fc27267fb503d8 Mon Sep 17 00:00:00 2001
From: Harikrishna Shenoy <h-shenoy@ti.com>
Date: Fri, 20 Jun 2025 17:29:21 +0530
Subject: [PATCH] Revert "drm/bridge: cdns-dsi-core: Remove empty FIFO check in
 cdns_dsi_check_conf()"

commit 49aa34cefedf ("drm/bridge: cdns-dsi-core:
Remove empty FIFO check in cdns_dsi_check_conf()")
The issue addressed by the reverted patch are included in
the series posted upstream.

Signed-off-by: Harikrishna Shenoy <h-shenoy@ti.com>
---
 drivers/gpu/drm/bridge/cadence/cdns-dsi-core.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/drivers/gpu/drm/bridge/cadence/cdns-dsi-core.c b/drivers/gpu/drm/bridge/cadence/cdns-dsi-core.c
index ee867b69a9d1..ae7da8098416 100644
--- a/drivers/gpu/drm/bridge/cadence/cdns-dsi-core.c
+++ b/drivers/gpu/drm/bridge/cadence/cdns-dsi-core.c
@@ -604,6 +604,17 @@ static int cdns_dsi_check_conf(struct cdns_dsi *dsi,
 	if (output->dev->mode_flags & MIPI_DSI_MODE_VIDEO_SYNC_PULSE)
 		dsi_hss_hsa_hse_hbp += dsi_cfg->hsa + DSI_HSA_FRAME_OVERHEAD;
 
+	/*
+	 * Make sure DPI(HFP) > DSI(HSS+HSA+HSE+HBP) to guarantee that the FIFO
+	 * is empty before we start a receiving a new line on the DPI
+	 * interface.
+	 */
+	if ((u64)phy_cfg->hs_clk_rate *
+	    mode_to_dpi_hfp(mode, mode_valid_check) * nlanes <
+	    (u64)dsi_hss_hsa_hse_hbp *
+	    (mode_valid_check ? mode->clock : mode->crtc_clock) * 1000)
+		return -EINVAL;
+
 	return 0;
 }
 
-- 
2.50.1

